### PR TITLE
zh-CN: sync the translated content of section `:host-context()`

### PR DESCRIPTION
**Pull request by daiwanxing**

### CODE REVIEW

File: files/zh-cn/web/css/_colon_host-context/index.md [hunk 1]
```diff
@@ -1,63 +1,79 @@
...
+```css-nolint
+:host-context(<compound-selector>) {
+  /* ... */
+}
+```
 
-## Examples
+## 示例
 
 下边的代码片段来自 [host-selectors](https://github.com/mdn/web-components-examples/tree/master/host-selectors) ([这里查看样式效果](https://mdn.github.io/web-components-examples/host-selectors/))。
```
Comment (skyclouds2001) [inline files/zh-cn/web/css/_colon_host-context/index.md:46]:
```suggestion
下边的代码片段来自 [host-selectors](https://github.com/mdn/web-components-examples/tree/master/host-selectors)（[这里查看样式效果](https://mdn.github.io/web-components-examples/host-selectors/)）。
```

File: files/zh-cn/web/css/_colon_host-context/index.md [hunk 2]
```diff
@@ -1,63 +1,79 @@
...
 
 在本例中，我们有一个简单的自定义元素 — `<context-span>` — 你可以把文字包裹在该自定义标签中：
```
Comment (skyclouds2001) [inline files/zh-cn/web/css/_colon_host-context/index.md:48]:
```suggestion
在本例中，我们有一个简单的自定义元素 —— `<context-span>` —— 你可以把文字包裹在该自定义标签中：
```

File: files/zh-cn/web/css/_colon_host-context/index.md [hunk 3]
```diff
@@ -1,63 +1,79 @@
...
 
 ```html
-<h1>Host selectors <a href="#"><context-span>example</context-span></a></h1>
+<h1>
+  Host selectors <a href="#"><context-span>example</context-span></a>
+</h1>
 ```
 
 在自定义元素的构造器函数中，我们创建 `style` 和 `span` 元素。让 `span` 里呈现的是自定义元素的文字内容，并且给`style` 元素一些 CSS 规则。
```
Comment (skyclouds2001) [inline files/zh-cn/web/css/_colon_host-context/index.md:56]:
```suggestion
在自定义元素的构造器函数中，我们创建 `style` 和 `span` 元素。让 `span` 里呈现的是自定义元素的文字内容，并且给 `style` 元素一些 CSS 规则。
```

File: files/zh-cn/web/css/_colon_host-context/index.md [hunk 4]
```diff
@@ -1,63 +1,79 @@
...
-title: ':host-context()'
+title: ":host-context()"
 slug: Web/CSS/:host-context
 ---
 
-{{CSSRef}}{{SeeCompatTable}}
+{{CSSRef}}
 
-**`:host-context()`** [CSS 伪类](/zh-CN/docs/Web/CSS/Pseudo-classes)的作用是选择 shadow DOM 中 shadow host，这个伪类内可以写关于该 shadow host 的 CSS 规则。 (我们可以从 shadow DOM 中选择一个自定义元素 custom element) — 但前提是，在 DOM 层级中，括号中的选择器参数必须和 shadow host 的祖先相匹配。
+**`:host-context()`** [CSS](/zh-CN/docs/Web/CSS) [伪类](/zh-CN/docs/Web/CSS/Pseudo-classes)函数选择包含其所在的 CSS 的 [shadow DOM](/zh-CN/docs/Web/API/Web_components/Using_shadow_DOM) 的影子宿主（因此你可以从其影子 DOM 内部选择自定义元素）—— 但前提是作为函数参数的选择器与影子宿主的祖先在 DOM 层次结构中的位置匹配。
```
Comment (yin1999) [inline files/zh-cn/web/css/_colon_host-context/index.md:8]: see also: #13432

File: files/zh-cn/web/css/_colon_host-context/index.md [hunk 5]
```diff
@@ -1,63 +1,79 @@
...
-title: ':host-context()'
+title: ":host-context()"
 slug: Web/CSS/:host-context
 ---
 
-{{CSSRef}}{{SeeCompatTable}}
+{{CSSRef}}
 
-**`:host-context()`** [CSS 伪类](/zh-CN/docs/Web/CSS/Pseudo-classes)的作用是选择 shadow DOM 中 shadow host，这个伪类内可以写关于该 shadow host 的 CSS 规则。 (我们可以从 shadow DOM 中选择一个自定义元素 custom element) — 但前提是，在 DOM 层级中，括号中的选择器参数必须和 shadow host 的祖先相匹配。
+**`:host-context()`** [CSS](/zh-CN/docs/Web/CSS) [伪类](/zh-CN/docs/Web/CSS/Pseudo-classes)函数选择包含其所在的 CSS 的 [影子 DOM（shadow DOM）](/zh-CN/docs/Web/API/Web_components/Using_shadow_DOM) 的影子宿主（shadow Host）（因此你可以从其影子 DOM 内部选择自定义元素）—— 但前提是作为函数参数的选择器与影子宿主的祖先在 DOM 层次结构中的位置匹配。
```
Comment (yin1999) [inline files/zh-cn/web/css/_colon_host-context/index.md:8]:
```suggestion
**`:host-context()`** [CSS](/zh-CN/docs/Web/CSS) [伪类](/zh-CN/docs/Web/CSS/Pseudo-classes)函数选择内部使用了该 CSS 的[影子 DOM（shadow DOM）](/zh-CN/docs/Web/API/Web_components/Using_shadow_DOM)的影子宿主（shadow host），因此你可以从其影子 DOM 内部选择自定义元素——但前提是作为函数参数的选择器与影子宿主的祖先在 DOM 层次结构中的位置匹配。
```

File: files/zh-cn/web/css/_colon_host-context/index.md [hunk 6]
```diff
@@ -1,63 +1,79 @@
...
-典型的使用方法是后代选择器表达式。例如 `h1` — 只选择在`<h1>` 内的自定义元素的实例。
+换句话说，这允许自定义元素或其影子 DOM 内的任何内容根据其在外部 DOM 中的位置或应用于祖先元素的类/属性应用不同的样式。
 
-> **备注：** 该伪类的 css 样式在 shadow DOM 之外的元素上是没有效果的
+这样子做的一个典型用法是与后代选择器表达式（例如 h1）一起使用，仅选择在 `<h1>` 内的自定义元素的实例。另一个典型用法是允许内部元素对任何祖先元素上的类或属性做出反应，例如，在将 `.dark-theme` 类应用于 `<body>` 时应用不同的文本颜色。
 
-```css
-/* 选择了一个 shadow root host，当且仅当这个
- shadow root host 是括号中选择器参数 (h1) 的后代 */
+> **备注：** 这不适用于在影子 DOM 之外使用。
```
Comment (yin1999) [inline files/zh-cn/web/css/_colon_host-context/index.md:14]:
```suggestion
> **备注：** 这在影子 DOM 之外使用时无效。
```

File: files/zh-cn/web/css/_colon_host-context/index.md [hunk 7]
```diff
@@ -1,63 +1,79 @@
...
 
+```css
+/* 选择 shadow root 宿主，仅当它是给定的选择器参数的后代 */
```
Comment (yin1999) [inline files/zh-cn/web/css/_colon_host-context/index.md:17]:
```suggestion
/* 选择影子根宿主，仅当它是给定的选择器参数的后代 */
```

File: files/zh-cn/web/css/_colon_host-context/index.md [hunk 8]
```diff
@@ -1,63 +1,79 @@
...
+:host-context(<compound-selector>) {
+  /* ... */
+}
+```
 
-## Examples
+## 示例
 
-下边的代码片段来自 [host-selectors](https://github.com/mdn/web-components-examples/tree/master/host-selectors) ([这里查看样式效果](https://mdn.github.io/web-components-examples/host-selectors/))。
+下边的代码片段来自 [host-selectors](https://github.com/mdn/web-components-examples/tree/master/host-selectors)（[这里查看样式效果](https://mdn.github.io/web-components-examples/host-selectors/)）。
```
Comment (yin1999) [inline files/zh-cn/web/css/_colon_host-context/index.md:46]:
```suggestion
下边的代码片段来自 [host-selectors 示例](https://github.com/mdn/web-components-examples/tree/master/host-selectors)（[也可以在线查看](https://mdn.github.io/web-components-examples/host-selectors/)）。
```

File: files/zh-cn/web/css/_colon_host-context/index.md [hunk 9]
```diff
@@ -1,63 +1,79 @@
...
 
-在本例中，我们有一个简单的自定义元素 — `<context-span>` — 你可以把文字包裹在该自定义标签中：
+在本例中，我们有一个简单的自定义元素 —— `<context-span>` —— 你可以把文字包裹在该自定义标签中：
```
Comment (yin1999) [inline files/zh-cn/web/css/_colon_host-context/index.md:48]:
```suggestion
在本例中，我们有一个简单的自定义元素（`<context-span>` ），你可以把文字包裹在该自定义标签中：
```

File: files/zh-cn/web/css/_colon_host-context/index.md [hunk 10]
```diff
@@ -1,63 +1,79 @@
...
+  "span:hover { text-decoration: underline; }" +
+  ":host-context(h1) { font-style: italic; }" +
+  ':host-context(h1):after { content: " - no links in headers!" }' +
+  ":host-context(article, aside) { color: gray; }" +
+  ":host(.footer) { color : red; }" +
+  ":host { background: rgba(0,0,0,0.1); padding: 2px 5px; }";
 ```
 
-`:host-context(h1) { font-style: italic; }` 和 `:host-context(h1):after { content: " - no links in headers!"` 这些 CSS 规则规定了位于`<h1>` 标签内部的 `<context-span>` 元素的实例的样式。
+`:host-context(h1) { font-style: italic; }` 和 `:host-context(h1):after { content: " - no links in headers!" }` 规则应用 `<h1>` 内部的 `<context-span>` 元素实例（在此实例中为影子宿主）。我们使用它来明确指出，在我们的设计中，自定义元素不应出现在 `<h1>` 内部。
```
Comment (yin1999) [inline files/zh-cn/web/css/_colon_host-context/index.md:76]:
```suggestion
`:host-context(h1) { font-style: italic; }` 和 `:host-context(h1):after { content: " - no links in headers!" }` 规则应用 `<h1>` 内部的 `<context-span>` 元素实例（在此实例中为影子宿主）。我们使用它来明确指出：在我们的设计中，自定义元素不应出现在 `<h1>` 内部。
```

**Review by yin1999 — APPROVED** [review]
Thank you @daiwanxing

**Pull request merged**